src/azul/portal_service.py

```diff
@@ -0,0 +1,82 @@
+from copy import deepcopy
+from typing import (
+    Sequence,
+)
+import json
+import logging
+
+import boto3
+from botocore.exceptions import ClientError
+
+from azul import config
+from azul.plugin import Plugin
+from azul.types import JSON
+
+log = logging.getLogger(__name__)
+
+
+class PortalService:
+
+    def get_portal_integrations_db(self) -> Sequence[JSON]:
+        """
+        Retrieve portal DB from S3.
+
+        If no object is present at the location specified in azul.config, then
+        a default hardcoded DB is obtained from the current plugin, uploaded to
+        S3, and returned.
+        """
+
+        client = boto3.client('s3')
+        try:
+            retrieval = client.get_object(Bucket=config.portal_integrations_db_bucket,
```

Comment:
```suggestion
            response = client.get_object(Bucket=config.portal_integrations_db_bucket,
```

is what we typically call this.